lib/storage/bucket.js

```diff
@@ -349,6 +351,105 @@ Bucket.prototype.setMetadata = function(metadata, callback) {
 };
 
 /**
+ * Make the bucket listing and optionally its contents publicly readable to
+ * anybody.
+ * NOTE: This function may be long-running and use a high number of requests as
+ * it recursively applies makePublic to every file within the bucket.
+ *
+ * @param {object=}  options - The configuration object.
+ * @param {boolean}  options.recursive - Whether to recursively apply makePublic
+ *     to every file in the bucket as well. Default is false.
+ * @param {boolean}  options.force - By default, makePublic will return at the
+ *     first error found, but set this to true to continue processing if this
+ *     occurs.
+ * @param {Function} callback - The callback function.
+ *
+ * @example
+ * bucket.makePublic({ recursive: true }, function(err, files) {
+ *   // err is the first error to occur, otherwise null.
+ *   // files is an array of files successfully made public in the bucket.
+ * });
+ */
+Bucket.prototype.makePublic = function(options, callback) {
+  var self = this;
+  var errors = [];
+  var filesProcessed = [];
+  var isDone = false;
+  var nextQuery = {};
+
+  if (util.is(options, 'function')) {
+    callback = options;
+    options = {};
+  }
+
+  // Allow public to read bucket contents
+  // while preserving original permissions
+  this.acl.add({
+    entity: 'allUsers',
+    role: 'READER'
+  }, function(err) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    if (options.recursive) {
+      async.whilst(function() {
+        // while there are still more files to make public
+        return !isDone && nextQuery;
+      }, function(cb) {
+        processMoreFiles(nextQuery, cb);
+      }, function(err) {
+        if (err || errors.length) {
+          callback(err || errors, filesProcessed);
+        } else {
+          callback(null, filesProcessed);
+        }
+      });
+    } else {
+      callback(null, filesProcessed);
+    }
+  });
+
+  function processMoreFiles(query, callback) {
+    self.getFiles(query, function(err, files, nextPageQuery) {
+      if (err) {
+        callback(err);
+        isDone = true;
+        return;
+      }
+
+      // Iterate through each file and make it public.
+      async.eachLimit(files, 10, function(file, cb) {
+        file.makePublic(function(err) {
+          if (err) {
+            if (!options.force) {
+              cb(err);
+            } else {
+              errors.push(err);
+              cb();
```

Comment:
This could probably use some more inline docs, but basically add errors to an array when they explicitly want to `force` to keep processing.